pdf/core/primitives.go

```diff
@@ -432,8 +484,7 @@ func (array *PdfObjectArray) GetAsFloat64Slice() ([]float64, error) {
 	slice := []float64{}
 
 	for _, obj := range array.Elements() {
-		obj := TraceToDirectObject(obj)
-		number, err := getNumberAsFloat(obj)
+		number, err := GetNumberAsFloat(TraceToDirectObject(obj))
```

Comment:
Possibly include tracing in GetNumberAsFloat,
although that means that some PDF is storing a number in an indirect object, which seems odd, but I am sure that there are some files that do it.